packages/clerk-js/src/ui/userProfile/connectedAccounts/ConnectedAccountList.tsx

```diff
@@ -1,34 +1,56 @@
 import { List } from '@clerk/shared/components/list';
 import { TitledCard } from '@clerk/shared/components/titledCard';
+import { OAuthProvider } from "@clerk/types";
 import React from 'react';
-import { useCoreUser } from 'ui/contexts';
+import { useCoreUser, useEnvironment } from 'ui/contexts';
 import { PageHeading } from 'ui/userProfile/pageHeading';
 
 import { ConnectedAccountListItem } from './ConnectedAccountListItem';
+import { UnconnectedAccountListItem } from './UnconnectedAccountListItem';
 
 export function ConnectedAccountList(): JSX.Element {
+  return (
+    <>
+      <PageHeading title='Connected accounts' backTo='./../'/>
+      <TitledCard className='cl-themed-card cl-list-card'>
+        <ConnectedAccountListRows/>
+      </TitledCard>
+    </>
+  );
+}
+
+function ConnectedAccountListRows(): JSX.Element {
   const user = useCoreUser();
+  const { userSettings: {social} } = useEnvironment();
+
+  const availableProviders = Object.values(social).filter(oauthProvider => oauthProvider.enabled);
+
+  const connectedProviders = user.externalAccounts.map(externalAccount => externalAccount.provider);
 
-  const buildExternalAccountRows = () =>
-    user.externalAccounts.map(externalAccount => (
+  const unconnectedProviders = availableProviders.filter((oauthProvider) => {
+    const provider = oauthProvider.strategy.replace('oauth_', '') as OAuthProvider; // :-(
+    return !connectedProviders.includes(provider);
+  });
+
+  if (availableProviders.length == 0) {
```

Comment:
🔧 If the instance doesn't have any enabled OAuth accounts let's skip the section completely. This is similar to how we handle usernames, phones, etc...

Reply:
Sure